org.adempiere.base/src/org/compiere/model/GridTable.java

```diff
@@ -3294,19 +3357,40 @@ private void doRun() {
 			fireDataStatusIEvent("", "");
 		}
 
+		/**
+		 *	Set Parameters from MQuery
+		 *  @param pstmt prepared statement
+		 *  @param query
+		 *  @return next position
+		 */
+		private int setParameter(PreparedStatement pstmt, MQuery m_query) {
+			int pos = 1;
+			if (m_query != null) {
+				for (Object param : m_query.getParameters()) { 
+					try {
+						pstmt.setObject(pos, param);
+					} catch (SQLException e) {
+						log.log(Level.SEVERE, "parameter", e);
+					}
+					pos++;
+				}
+			}
+			return pos;
+		}
+
 		/**
 		 *	Set Parameter for Query.
 		 *		elements must be Integer, BigDecimal, String (default)
 		 *  @param pstmt prepared statement
 		 *  @param countSQL count
 		 */
-		private void setParameter (PreparedStatement pstmt, boolean countSQL)
+		private void setParameter (PreparedStatement pstmt, boolean countSQL, int nextPos)
 		{
 			if (m_parameterSELECT.size() == 0 && m_parameterWHERE.size() == 0)
 				return;
 			try
 			{
-				int pos = 1;	//	position in Statement
+				int pos = nextPos;	//	position in Statement
 				//	Select Clause Parameters
 				for (int i = 0; !countSQL && i < m_parameterSELECT.size(); i++)
 				{

```

Comment:
the existing GridTable actually has support for parameters (not sure though whether it is fully working) - the setParameterSelect and setParameterWhere methods. To move to the use of parameters, we should either complete the implementation using the 2 methods above or if it is too broken, we should remove those 2 method as it will make the GridTable API very confusing.